### PR TITLE
New exercises

### PR DESCRIPTION
**Pull request by jrmlhermitte**
I updated the exercise a bit with hints for converting the other values. Note, I did not notice but there was something missing in the conversion code before here it is. good luck!

### CODE REVIEW

**Comment by elParcero** [comment]
Alright thank you, I appreciate it !

File: exercises/06-iss-file-handler.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    this to open and close the files for us. More on that later.
+
+'''
+
+class ANReader:
+    def __init__(self, resource_path, chunk_size=1024):
+        # load the file or save the filepath
+        pass
+
+    def __call__(self, chunk_num):
```
Comment (jrmlhermitte) [inline exercises/06-iss-file-handler.py]: we can use `row_num` first for now.

File: exercises/04-iss-read-raw-data.py [hunk 2]
```diff
@@ -53,27 +53,26 @@
...
                 rather than by hand. (But don't worry if it's too tedious)
 
+    Another hint:
+        Try to see if you can use:
+            pd.read_csv to read the files
+            and try to see if you can use "apply" to apply a transformation to
+            a file
+
+            for example:
+                df['encoder'] = df['encoder'].apply(enc2counts)
```
Comment (jrmlhermitte) [inline exercises/04-iss-read-raw-data.py]: try this for the `encoder` and `count` rows

File: exercises/04-iss-read-raw-data.py [hunk 3]
```diff
@@ -53,27 +53,26 @@
...
-        adc = adcs[i]
-        shifted_adc = adc >> 8
-        if shifted_adc > 0x1FFFF:
-            count = (shifted_adc) - 0x40000
-        else:
-            count = (shifted_adc)*7.62939453125e-05
-
-    return counts
+fc = 7.62939453125e-05
+adc2counts = lambda x: ((int(x, 16) >> 8) - 0x40000) * fc \
```
Comment (jrmlhermitte) [inline exercises/04-iss-read-raw-data.py]: this is same as:
```py3
def myfunc(x):
    if (int(x, 16) >> 8) > 0x1FFFF:
        ((int(x, 16) >> 8) - 0x40000) * fc
    else:
       (int(x, 16) >> 8)*fc
```

File: exercises/04-iss-read-raw-data.py (hunk 2 above)
Comment (jrmlhermitte) [reply exercises/04-iss-read-raw-data.py]:
```py3
df['totaltime'] = df['time (s)'] + 1e-9*df['time (ns)']
```

**Pull request merged**